Source/AwsCommonRuntimeKit/http/HTTPClientConnectionManagerCallbackCore.swift

```diff
@@ -39,13 +39,28 @@ private func onConnectionSetup(connection: UnsafeMutablePointer<aws_http_connect
                                errorCode: Int32,
                                userData: UnsafeMutableRawPointer!) {
     let callbackDataCore = Unmanaged<HTTPClientConnectionManagerCallbackCore>.fromOpaque(userData!).takeRetainedValue()
+    let continuation = callbackDataCore.continuation
+
     if errorCode != AWS_OP_SUCCESS {
-        callbackDataCore.continuation.resume(throwing: CommonRunTimeError.crtError(CRTError(code: errorCode)))
+        continuation.resume(throwing: CommonRunTimeError.crtError(CRTError(code: errorCode)))
         return
     }
 
     // Success
-    let httpConnection = HTTPClientConnection(manager: callbackDataCore.connectionManager,
-                                              connection: connection!)
-    callbackDataCore.continuation.resume(returning: httpConnection)
+    switch aws_http_connection_get_version(connection) {
+    case AWS_HTTP_VERSION_2: continuation.resume(
+        returning: HTTP2ClientConnection(
+            manager: callbackDataCore.connectionManager,
+            connection: connection!))
+    case AWS_HTTP_VERSION_1_1:
+        continuation.resume(
+            returning: HTTPClientConnection(
+                manager: callbackDataCore.connectionManager,
+                connection: connection!))
+    default:
```

Comment:
default should be avoided per Swift best practices, you can add unknown here and do the right think. I suppose we want to throw an error or log something.

Reply:
I don't think we can use unkown default in this case. As per https://docs.swift.org/swift-book/ReferenceManual/Attributes.html#ID620, this attribute is only allowed when compiling with library evolution mode. It is a C enum, even if I handle all the cases, it still says "switch must be exhaustive, remove '@unknown' to handle remaining values". We are already throwing an error.

Reply:
ah my bad, it is C enum. Thanks for pointing that out, all good.